Straight_skeleton_2/doc/Straight_skeleton_2/CGAL/extrude_skeleton.h

```diff
@@ -0,0 +1,94 @@
+namespace CGAL {
+
+/**
+* \ingroup PkgStraightSkeleton2Extrusion
+*
+* \brief constructs the straight skeleton-based extrusion of a polygon with holes.
+*
+* Given a polygon with holes and a set of weights, the skeleton extrusion is a volume constructed
+* from the weighted straight skeleton by associating a height to the vertices of the skeleton,
+* which corresponds to the time at the vertex. The input polygon is placed at `z = 0`.
+*
+* This function allows cropping the extruded skeleton at a maximum height, using the optional
+* `maximum_height` named parameter.
+*
+* The result is a closed, 2-manifold surface triangle mesh. Note that this mesh can have non-local
+* self-intersections in some case of cropping.
+*
+* @tparam PolygonWithHoles must be a model of `SequenceContainer` with value type `InK::Point_2` (e.g. `Polygon_2<InK>`)
+                           or a model of `GeneralPolygonWithHoles_2` (e.g. `Polygon_with_holes_2<InK>`).
+* @tparam PolygonMesh a model of `MutableFaceGraph`
+* @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+*
+* @param polygon the polygon with holes
+* @param out the output polygon mesh
+* @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+*
+* \cgalNamedParamsBegin
+*   \cgalParamNBegin{weights}
+*     \cgalParamDescription{a range of weights, one for each edge of the input polygon}
+*     \cgalParamType{a model of `Range` whose value type is a model of `Range` whose value type is `FT`}
+*     \cgalParamDefault{an empty range (uniform weights are used)}
+*     \cgalParamExtra{Weights should be finite and all of the same sign.
+*                     Negative weights are used to signify exterior extrusion.
+*                     A weight `0` can exceptionally be used to signify a vertical extrusion of this edge.}
```

Comment:
*exceptionally* what does that mean ?

Reply:
It means usually you are not allowed to give a 0 weight (that's the "strictly positive"), but here it is tolerated as a way of asking for a vertical slab. I will rephrase.